### PR TITLE
Switch to Java 11's HttpClient

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ CHANGELOG
 * On `HttpException` and `InvalidRequestException`, `getUrl()` has been
   deprecated in favor of `getUri()`. Constructors that took a `URL` have
   been replaced with the equivalent taking a `URI`.
+* Deprecated constructors on model an trait classes were removed.
```
Comment (horgh) [inline CHANGELOG.md:20]: "and"

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+  * `connectTimeout(int)` has been deprecated in favor of
+    `connectTimeout(Duration)`.
+  * `readTimeout(int)` has been deprecated in favor of
+    `requestTimeout(Duration)`.
+  * `proxy(Proxy)` has been deprecated in favor of `proxy(ProxySelector)`.
+* On `HttpException` and `InvalidRequestException`, `getUrl()` has been
+  deprecated in favor of `getUri()`. Constructors that took a `URL` have
+  been replaced with the equivalent taking a `URI`.
+
+
```
Comment (horgh) [inline CHANGELOG.md:22]: Stray newline!

File: src/main/java/com/maxmind/geoip2/WebServiceClient.java [hunk 3]
```diff
@@ -237,17 +237,42 @@ public Builder locales(List<String> val) {
...
+        /**
+         * @param val Request timeout duration. The default is 20 seconds.
+         * @return Builder object
+         */
+        public Builder requestTimeout(Duration val) {
+            this.requestTimeout = val;
+            return this;
+        }
+
+
```
Comment (horgh) [inline src/main/java/com/maxmind/geoip2/WebServiceClient.java:257]: Stray newline

**Review by horgh — CHANGES_REQUESTED** [review]
Sweet!

File: src/main/java/com/maxmind/geoip2/record/Traits.java [hunk 4]
```diff
@@ -45,195 +45,17 @@
...
-
-    public Traits(String ipAddress) {
-        this(null, null, null, ipAddress, false, false, null, null, null);
+        this(null, null, null, null,
+                null, false, false, false, false,
+                false, false, false, false, false, null,
+                null, null, null, null, null, null, null);
     }
 
     public Traits(String ipAddress, Network network) {
```
Comment (horgh) [inline src/main/java/com/maxmind/geoip2/record/Traits.java:54]: What's using this one? Actually, what uses the no argument variants too?

Comment (oschwald) [reply src/main/java/com/maxmind/geoip2/record/Traits.java:54]: Hmm, I am not entirely sure. I think Jackson used to use the no argument ones, but we switched how we use it. Maybe they can be removed too. I'll try.

Comment (oschwald) [reply src/main/java/com/maxmind/geoip2/record/Traits.java:54]: This and the no-argument ones are used from the constructors for the model classes as we guarantee that things like `.getTraits()` will return an object even if there was no traits data in the response.

**Review by horgh — APPROVED** [review]

**Pull request merged**